### ISSUE TITLE
[WASM] DateTime.AddDays/AddSeconds/AddMinutes not fully working if project configured for Uno.SkiaSharp

### ISSUE THREAD

**Issue by peternary** [unoplatform/uno#1890]
## Current behavior

If you have '<MonoWasmRuntimeConfiguration>release-dynamic</MonoWasmRuntimeConfiguration>' set in the .csproj file for a wasm project, as specified [here](https://github.com/unoplatform/Uno.SkiaSharp), then using a negative number in DateTime.AddDays returns the DateTime object's current value, not an updated value.  DateTime.AddDays with a positive number works as expected.  Similar behavior exhibited by AddMinutes, AddHours, and AddSeconds.  Behavior does not appear to be affected by the actual presence of Uno.SkiaSharp nuget packages.

Note: the AOT build _does_ work properly.

## Expected behavior

DateTime.AddDays (and related functions) should properly calculate negative values.

## How to reproduce it (as minimally and precisely as possible)

In the attached sample, load the Uwp build and observe that the dates are yesterday, today, and tomorrow.  Load WASM and observe that the dates are today, today, and tomorrow.  In TestMonoWasmConfiguration.Wasm.csproj remove the line '\<MonoWasmRuntimeConfiguration>release-dynamic\</MonoWasmRuntimeConfiguration>' and rebuild  WASM - observe that the dates are yesterday, today, and tomorrow.

[TestMonoWasmConfiguration.zip](https://github.com/unoplatform/uno/files/3744300/TestMonoWasmConfiguration.zip)

## Environment

Nuget Package: 
Uno.UI

Package Version(s):
1.0.0-dev.302

Affected platform(s):

- [ ] iOS
- [ ] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

Visual Studio:

- [x] 2017 (version: )
- [x] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins:

- [ ] Resharper (version: )

**Comment by jeromelaban** [comment]
Thanks Peter for the detailed report!

This looks like a mono-wasm issue, and at this point, the only option we have is to update the runtime to a later version to fix it.

At present time, the update is blocked on https://github.com/unoplatform/Uno.Wasm.Bootstrap/issues/114, which also blocks #1451. This issue is definitely a different one, so we'll track it as a different one, as at this time, the mono-wasm latest release-dynamic runtime crashes a simple `Console.WriteLine`.

**Comment by DierkDroth** [comment]
@jeromelaban would you have any idea on when the next mono-wasm update would be due?

**Comment by jeromelaban** [comment]
@DierkDroth multiple issues need to be fixed first:
- https://github.com/mono/mono/issues/16818, which is blocked on https://github.com/jbevain/cecil/pull/622#issuecomment-336372576
- https://github.com/WebAssembly/binaryen/issues/2180

The first one I fast tracked by fixing it, it's waiting on validation, the second one is pending, but I do not know when I'll be available. This current issue will be a third one that needs fixing.

So to answer your question, there's no due time at the moment, unfortunately.

**Comment by DierkDroth** [comment]
@jeromelaban thanks for sharing that info.

**Comment by DierkDroth** [comment]
@jeromelaban the issues on links you provided above appear to be closed (pls correct me if I'm wrong). Does this mean work on this very issue could be resumed?

**Comment by jeromelaban** [comment]
@DierkDroth there is another emscripten issue, related to https://github.com/unoplatform/Uno.Wasm.Bootstrap/issues/128, which is the exact issue this bug is about. The current runtime fails to perform the computation (as you've seen), yet silently, and the latest emscripten fails explicitly.

Once this issue is fixed, which should be soon, as there's activity there in the past few days, the work will resume.

**Comment by DierkDroth** [comment]
Thanks for the update @jeromelaban

**Comment by jeromelaban** [comment]
Update: Another issue needs to be fixed https://github.com/emscripten-core/emscripten/issues/9850

**Comment by DierkDroth** [comment]
Shoot ... thanks for the update @jeromelaban

**Comment by jeromelaban** [comment]
It's almost fixed, though, should be quick :)

**Comment by DierkDroth** [comment]
Appears emscripten-core/emscripten#9850 is fixed now. Any chance that the original issue could be addressed?

**Comment by jeromelaban** [comment]
@DierkDroth it is fixed, but not published yet unfortunately... There's not release date but the activity suggests there may be one soon.

**Comment by DierkDroth** [comment]
Thanks for the update @jeromelaban

**Comment by jeromelaban** [comment]
Emscripten has been updated, an update to mono has been submitted : https://github.com/mono/mono/pull/18016

**Comment by jeromelaban** [comment]
There's another update that needs to happen in emscripten (https://github.com/emscripten-core/emscripten/issues/9950), we'll update mono in parallel. This emscripten issue does not touch Uno directly, but prevents the compilation of Skia.

**Comment by DierkDroth** [comment]
Thanks for the update @jeromelaban. Looks like a never ending story ...

**Comment by DierkDroth** [comment]
@jeremienventive would it make sense to poke the right people in order to see some progress on the pending issues so we had a chance to get this very issue resolved?

**Comment by jeromelaban** [comment]
@DierkDroth we're still waiting for an update on the binaryen side, so we do not have control over the priority of that issue, unfortunately. It looks that a few other people are waiting on that specific issue as well, so it may move faster.

**Comment by DierkDroth** [comment]
Thanks for your feedback @jeromelaban 

I know it's not your/nventive's responsibility nevertheless I wanted to share the feeling that things are moving way slower than I'd hoped on the mono-wasm support side of tings.

**Comment by jeromelaban** [comment]
@DierkDroth On this particular topic, indeed it's slower than we would have hoped. In general, we can influence issues by contributing but in this case, this is a very deep issue and it's the third try to fix it from the original author. Upvoting the issue on emscripten may help as well for visibility.

**Comment by DierkDroth** [comment]
>> Upvoting the issue on emscripten may help as well for visibility.

I'm sorry, I'm too far off the tech side of things here: how/where could I upvote?

**Comment by jeromelaban** [comment]
@DierkDroth Adding a pair of eyes or a thumbs up:
![image](https://user-images.githubusercontent.com/5839577/72274251-ccf5a180-35f9-11ea-9c2b-51ba67e490b0.png)

**Comment by DierkDroth** [comment]
@jeromelaban looks like the relevant merges took place. Any chance to get the original issue addressed?

**Comment by jeromelaban** [comment]
@DierkDroth the merges have been made in Binaryen, but are not yet in an emscripten release. Once it is available, we'll be able to move forward.

**Comment by DierkDroth** [comment]
I see. Where could that be tracked/viewed?

**Comment by jeromelaban** [comment]
Releases are available there: https://github.com/emscripten-core/emsdk but there's no timetable for individual releases that I could find.

**Comment by DierkDroth** [comment]
Hmm, I saw no releases there but saw releases here https://github.com/emscripten-core/emscripten/releases

Is that what I needed to be looking for?

**Comment by jeromelaban** [comment]
Yes, those are the past releases, we'll most likely take 1.39.7 if the binaryen fixes stick.

**Comment by DierkDroth** [comment]
Got it. Thx

**Comment by DierkDroth** [comment]
@jeromelaban 1.39.7 appears to be out now. No idea if it would hod the relevant fix(es) though...

**Comment by jeromelaban** [comment]
Thanks, the update to 1.39.7 has passed in mono already https://github.com/mono/mono/pull/18687, we should be able to go further in the investigation.

**Comment by DierkDroth** [comment]
Excellent, thanks. Looking forward to see this issue closed.

**Comment by jeromelaban** [comment]
I've been testing the latest release of emscripten (1.39.7), and unfortunately there are other issues that have been introduced when using the dynamic linking mode.

I'm going to be closing this issue as it is fixed in the latest mono (as part of the 1.1.0-dev.409 bootstrapper version), but the emscripten updated does not fix the issues related to libSkia (https://github.com/unoplatform/uno/issues/1451). I'll provide updates for the bootstrapper changes there.

**Comment by peternary** [comment]
Issue tests as fixed for me.
